File: legacy-api/src/tasks/task.entity.ts
import { Prop, Schema, SchemaFactory } from "@nestjs/mongoose";
import { ObjectId } from "mongodb";
import { UserEntity } from "../users/user/user.entity";
import mongoose from "mongoose";

@Schema({ collection: 'tasks', timestamps: true })
export class TaskEntity {
    @Prop()
    _id: ObjectId;

    @Prop()
    title: string;

    @Prop(
        {
            type: mongoose.Schema.Types.ObjectId,
            ref: UserEntity.name
        }
    )
    user: UserEntity;

    @Prop()
    description?: string;

    @Prop()
    start_date?: string;

    @Prop()
    completed?: boolean;

    @Prop()
    created_at: Date;

    @Prop()
    updated_at: Date;
}

export const TaskEntitySchema = SchemaFactory.createForClass(TaskEntity);

// convert the _id to id
TaskEntitySchema.set('toJSON', {
    transform: (doc, ret) => {
        ret.id = ret._id;
        delete ret._id;
        delete ret.__v;
    }
});

File: api/models/task_entity.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type TaskEntity struct {
	ID		  string `json:"id" bson:"_id"`
	Title    string `json:"title" bson:"title" binding:"required"`
	User primitive.ObjectID `json:"user" bson:"user"`
	Description *string `json:"description" bson:"description"`
	StartDate *primitive.DateTime `json:"startDate" bson:"start_date" binding:"required"`
	EndDate *primitive.DateTime `json:"endDate" bson:"end_date" binding:"required"`
	Completed *bool `json:"completed" bson:"completed"`
	CreatedAt string `json:"createdAt" bson:"created_at"`
	UpdatedAt string `json:"updatedAt" bson:"updated_at"`
}